py/desispec/workflow/desi_proc_funcs.py

```diff
@@ -605,6 +611,120 @@ def get_desi_proc_tilenight_batch_file_pathname(night, tileid, reduxdir=None):
+
+    Returns:
+        scriptfile: the full path name for the script written.
+
+    Note:
+        batchdir and jobname can be used to define an alternative pathname, but may not work with assumptions in desi_proc.
+        These optional arguments should be used with caution and primarily for debugging.
+    """
+    jobdesc = 'linkcal'
+
+    if cameras is None or np.isscalar(cameras):
+        camword = cameras
+        cameras = decode_camword(camword)
+
+    if batchdir is None:
+        batchdir = get_desi_proc_batch_file_path(newnight)
+
+    os.makedirs(batchdir, exist_ok=True)
+
+    if jobname is None:
+        jobname = get_desi_proc_batch_file_name(night=newnight, exp="",
+                                                jobdesc=jobdesc, cameras=cameras)
+
+    if timingfile is None:
+        timingfile = f'{jobname}-timing-$SLURM_JOBID.json'
+        timingfile = os.path.join(batchdir, timingfile)
+
+    scriptfile = os.path.join(batchdir, jobname + '.slurm')
+
+    ## If system name isn't specified, pick it based upon jobdesc
+    if system_name is None:
+        system_name = batch.default_system(jobdesc=jobdesc)
+
+    batch_config = batch.get_config(system_name)
+    threads_per_core = batch_config['threads_per_core']
+    gpus_per_node = batch_config['gpus_per_node']
+    ncameras = len(cameras)
+
+
+    ncores, nodes, runtime = determine_resources(ncameras, jobdesc.upper(),
+                                                 forced_runtime=runtime,
+                                                 system_name=system_name)
+
+    runtime_hh = int(runtime // 60)
+    runtime_mm = int(runtime % 60)
+
+    with open(scriptfile, 'w') as fx:
+        fx.write('#!/bin/bash -l\n\n')
+        fx.write('#SBATCH -N {}\n'.format(nodes))
+        fx.write('#SBATCH --qos {}\n'.format(queue))
+        for opts in batch_config['batch_opts']:
+            fx.write('#SBATCH {}\n'.format(opts))
+        if batch_opts is not None:
+            fx.write('#SBATCH {}\n'.format(batch_opts))
+        if system_name == 'perlmutter-gpu':
+            # perlmutter-gpu requires projects name with "_g" appended
+            fx.write('#SBATCH --account desi_g\n')
+        else:
+            fx.write('#SBATCH --account desi\n')
+        fx.write('#SBATCH --job-name {}\n'.format(jobname))
+        fx.write('#SBATCH --output {}/{}-%j.log\n'.format(batchdir, jobname))
+        fx.write('#SBATCH --time={:02d}:{:02d}:00\n'.format(runtime_hh, runtime_mm))
+        #fx.write('#SBATCH --exclusive\n')
+
+        fx.write('\n')
+
+        fx.write(f'# {jobdesc} with {ncameras} cameras\n')
+        fx.write(f'# using {ncores} cores on {nodes} nodes\n\n')
+
+        fx.write('echo Starting at $(date)\n')
+
+        mps_wrapper=''
+        fx.write("export OMP_NUM_THREADS=1\n")
+
+        fx.write(f'\n# Link {refnight=} to {newnight=}\n')
+        srun=(f'srun -N {nodes} -n {ncores} -c {threads_per_core} '
+              + f'--cpu-bind=cores {cmd}')
```

Comment:
Since desi_link_calibnight is not MPI-parallelized, it does not need an srun prefix (which just adds extra overhead in this case).